security/pkg/pki/ca/controller/secret.go

```diff
@@ -207,9 +210,17 @@ func (sc *SecretController) upsertSecret(saName, saNamespace string) {
 		PrivateKeyID: key,
 		RootCertID:   rootCert,
 	}
-	_, err = sc.core.Secrets(saNamespace).Create(secret)
+
+	// We retry several times when create secret.
+	for i := 0; i < secretCreationRetry; i++ {
```

Comment:
Do you think it's better to add a delay between the retries?

Reply:
Changed it to sleep 1 second.